### ISSUE TITLE
Enable Link-Time Optimization (LTO)

### ISSUE THREAD

**Issue by zamazan4ik** [bnkc/emval#7]
Hi!

I noticed that in the `Cargo.toml` file Link-Time Optimization (LTO) for the project is not enabled. I suggest switching it on since it will reduce the binary size (always a good thing to have) and will likely improve the application's performance. If you want to read a bit more about LTO, I can recommend starting from [this](https://doc.rust-lang.org/rustc/codegen-options/index.html#lto) Rustc documentation.

I suggest enabling LTO only for the Release builds so as not to sacrifice the developers' experience while working on the project since LTO consumes an additional amount of time to finish the compilation routine. If you think that a regular Release build should not be affected by such a change as well, then I suggest adding an additional `dist` or `release-lto` profile where in addition to regular `release` optimizations LTO will also be added. Such a change simplifies life for maintainers and others interested in the project persons who want to build the most performant version of the package. If we enable it on the Cargo profile level, users, who build the package manually will get the LTO-optimized version "automatically". E.g., check `cargo-outdated` Release [profile](https://github.com/kbknapp/cargo-outdated/blob/master/Cargo.toml#L48).

Basically, it can be enabled with the following lines:

```
[profile.release]
lto = true
```

According to my quick local tests with the latest `emval` version on Fedora 41 + Rustc 1.82, adding `lto = true` to the Release profile reduces the final wheel size from 2.2 Mib to 1.8 Mib (built with the `maturin build --release` command).

Thank you.

**Comment by bnkc** [comment]
@zamazan4ik This seems sensible. Please feel free to open a PR and I will review. Thankyou!

**Comment by zamazan4ik** [comment]
Sure: https://github.com/bnkc/emval/pull/8
